pkg/ingester/shipper.go

```diff
@@ -286,18 +317,71 @@ func writeShipperMetaFile(logger log.Logger, dir string, meta *shipperMeta) erro
 	if err := f.Close(); err != nil {
 		return err
 	}
+
+	// Keep backwards compatibility with the previous Mimir version.
+	// TODO Remove in Mimir 2.7.0
+	thanosMeta := &thanosShipperMeta{Version: shipperMetaVersion1}
+	for _, shipped := range meta.Shipped {
+		thanosMeta.Uploaded = append(thanosMeta.Uploaded, shipped.BlockID)
+	}
+	if err := writeThanosShipperMetaFile(logger, dir, thanosMeta); err != nil {
+		return errors.Wrap(err, "writing thanos shipped meta file")
+	}
+
 	return renameFile(logger, tmp, path)
 }
 
-// readShipperMetaFile reads the given meta from <dir>/thanos.shipper.json.
-func readShipperMetaFile(dir string) (*shipperMeta, error) {
+// writeThanosShipperMetaFile writes the given meta into <dir>/thanos.shipper.json.
+func writeThanosShipperMetaFile(logger log.Logger, dir string, meta *thanosShipperMeta) error {
```

Comment:
`writeShipperMetaFile` and `writeThanosShipperMetaFile` have almost the same body, can we remove this duplication?

Reply:
`writeThanosShipperMetaFile` will get removed in a future release plus it also gets called from `writeShipperMetaFile`, I could substract the file creation meta encoding and os file creation to its own method but then it would only have one caller in the future